app/src/main/java/com/globalaccelerex/mpossdksample/ui/CheckoutFragment.kt

```diff
@@ -38,23 +62,45 @@ class CheckoutFragment : Fragment(R.layout.fragment_checkout) {
                 NumberFormat.getCurrencyInstance(Locale("en", "NG")).format(it)
         }
         fragmentClothBinding.payButton.setOnClickListener {
-            SweetAlertDialog(requireContext(), SweetAlertDialog.SUCCESS_TYPE)
-                .setTitleText("Success")
-                .setContentText("Payment Verified").apply {
-                    setCancelable(false)
-                    show()
-                    confirmText = requireContext().resources.getString(R.string.dialog_ok)
-                    setConfirmClickListener {
-//                        findNavController().navigate(R.id.parent)
-                        finishAffinity(requireActivity())
-                    }
-                }
-
+            startCheckout()
         }
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _fragmentClothBinding = null
     }
+
+    private fun showCheckoutResult(responseData: CardTransactionResponse) {
+        // Show result
+//        responseData.
+        SweetAlertDialog(requireContext(), SweetAlertDialog.SUCCESS_TYPE)
+            .setTitleText("Success")
+            .setContentText("Payment Verified").apply {
+                setCancelable(false)
+                show()
+                confirmText = requireContext().resources.getString(R.string.dialog_ok)
+                setConfirmClickListener {
+                    findNavController().navigate(CheckoutFragmentDirections.actionCheckoutFragmentToCatalogFragment())
+                    viewModel.clearSession()
+                    dismissWithAnimation()
+                }
+            }
+
+    }
+
+    private fun startCheckout() {
+        // Create a checkout request
+//        val checkoutAmount = 2.0
+        val request = PaymentRequest {
+            requestType = RequestType.PURCHASE
+//            amount = checkoutAmount
+            viewModel.totalPrice.observe(viewLifecycleOwner) {
```

Comment:
The total price is already set by now. It doesn't make sense to observe it. You can simply get the value here. `viewModel.totalPrice.value!!`

Reply:
Has been resolved Sir